src/main/java/org/jabref/logic/importer/fetcher/BiodiversityLibrary.java

```diff
@@ -0,0 +1,276 @@
+                }
+                // DOI
+                if (itemsDetails.has("Doi")) {
+                    entry.setField(StandardField.DOI, itemsDetails.getString("Doi"));
+                }
+                // Publisher
+                if (itemsDetails.has("PublisherName")) {
+                    entry.setField(StandardField.PUBLISHER, itemsDetails.getString("PublisherName"));
+                }
+                // Volume
+                if (itemsDetails.has("Volume") && !entry.hasField(StandardField.VOLUME)) {
+                    entry.setField(StandardField.VOLUME, itemsDetails.getString("Volume"));
+                }
+                // Date
+                if (itemsDetails.has("Date") && !entry.hasField(StandardField.DATE) && !entry.hasField(StandardField.YEAR)) {
+                    entry.setField(StandardField.DATE, itemsDetails.getString("Date"));
+                }
+                // Link
+                if (itemsDetails.has("PartUrl")) {
+                    entry.setField(StandardField.URL, itemsDetails.getString("PartUrl"));
+                }
+
+            }
+
+            if (item.getString("BHLType").equals("Item")) {
+                URL url = getItemMetadataURL(item.getString("ItemID"));
+                JSONObject itemsDetails = getDetails(url);
+                // Editor
+                if (itemsDetails.has("Sponsor")) {
+                    entry.setField(StandardField.EDITOR, itemsDetails.getString("Sponsor"));
+                }
+                // Publisher
+                if (itemsDetails.has("HoldingInstitution")) {
+                    entry.setField(StandardField.PUBLISHER, itemsDetails.getString("HoldingInstitution"));
+                }
+                // Language
+                if (itemsDetails.has("Language")) {
+                    entry.setField(StandardField.LANGUAGE, itemsDetails.getString("Language"));
+                }
+                // Link
+                if (itemsDetails.has("ItemUrl")) {
+                    entry.setField(StandardField.URL, itemsDetails.getString("ItemUrl"));
+                }
+                // Date
+                if (itemsDetails.has("Date") && !entry.hasField(StandardField.DATE) && !entry.hasField(StandardField.YEAR)) {
+                    entry.setField(StandardField.DATE, itemsDetails.getString("Date"));
+                }
+
+            }
+
+        }
+        return entry;
+    }
+
+    public BibEntry jsonResultToBibEntry(JSONObject item) {
+        BibEntry entry = new BibEntry();
+
+        if ("Book".equals(item.optString("Genre"))) {
+            entry.setType(StandardEntryType.Book);
+        } else {
+            entry.setType(StandardEntryType.Article);
+        }
+
+        // Title
+        if (item.has("Title")) {
+            entry.setField(StandardField.TITLE, item.optString("Title"));
+        }
+        // Authors
+        if (item.has("Authors")) {
+            JSONArray authors = item.getJSONArray("Authors");
```

Comment:
I think this can be improved, see for example: https://github.com/JabRef/jabref/blob/6e99a3390900a65a8d06b6dfa933889667a619f2/src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java#L149-L162